packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart

```diff
@@ -3,18 +3,14 @@
 // found in the LICENSE file.
 
 import 'dart:async';
-
 import 'package:flutter/foundation.dart';
-import 'package:flutter/services.dart';
 import 'package:meta/meta.dart';
+import 'package:firebase_analytics_platform_interface/firebase_analytics_platform_interface.dart';
 
-const MethodChannel firebaseChannel =
-    MethodChannel('plugins.flutter.io/firebase_analytics');
+final _platformInstance = FirebaseAnalyticsPlatform.instance;
```

Comment:
Move this inside the FirebaseAnalytics class (so it's a final instance variable and not a global...)

Reply:
(Then modify the constructor of FirebaseAnalyticsAndroid to receive a _platformInstance as a parameter, so you can pass the instance you already have into the FirebaseAnalyticsAndroid object)

Reply:
Done!